test/test.js

```diff
@@ -164,17 +164,17 @@ describe('vite', () => {
         )
         await expectByPolling(
           () => browserLogs[browserLogs.length - 1],
-          'js module hot updated:  /testHmrManual.js'
+          'js module hot updated:  /testHmrManualDep.js'
         )
         expect(
-          browserLogs.slice(browserLogs.length - 7, browserLogs.length - 1)
+          browserLogs.slice(browserLogs.length - 4, browserLogs.length - 1)
         ).toEqual([
           // dispose for both dep and self
-          `foo was: 2`,
+          // `foo was: 2`,
```

Comment:
This is actually an intentional fix because self-accepting callbacks **should** be called when a dep change bubbles up to the boundary. (e.g. https://github.com/vitejs/vite/issues/207)